### ISSUE TITLE
Enable Automatic Unlock Requests by Default

### ISSUE THREAD

**Issue by iandunn** [pfn/keepasshttp#209]
Allowing browser plugins to unlock KeePass (#67) makes a world of difference for convenience and user experience because it removes unnecessary manual steps and makes the process feel much more seamless.

I wasn't even aware that it was even a possibility, though, until I got frustrated enough with the clunky default process to start searching the web for solutions. I was pleasantly surprised to learn that it was actually possible, and is just disabled by default.

As far as I can remember, there wasn't any mention of it during the process of installing and configuring KeePass, KeePassHttp, and chromeIPass. After using 1Password for years, where the workflow and integration has been designed very well, it didn't even occur to me that this feature would be available but disabled.

I don't think it makes sense for a feature that is fundamental to a good user experience to be disabled by default, especially when it's not obvious that it exists and needs to be manually enabled.

What do you think about flipping it so that it just works out of the box?

**Comment by iandunn** [comment]
After having this feature enabled for a couple days, I ran into a pretty big problem: KeePass will be opened for any site that has a username/password field, even if I don't have a login stored for it, which is very annoying.

I think that's probably something that would need to be solved before this could be enabled by default. 

In the mean time, I've disabled the setting and changed my workflow to the following, which isn't perfect, but is pretty good:

1. Visit site I want to log in to 
1. Hit `ctrl+shift+F6` (or `ctrl+shift+U`), which opens KeePass
1. Enter master password
1. Allow the request (if it's the first time using that login)
1. Alt-tab back to the site and hit enter to submit the login form

**Comment by pfn** [comment]
while the database is locked, it is not possible to detect whether
credentials are available. (it's possible to cache results from previous
queries, but the less data stored in-browser, the better)

**Comment by iandunn** [comment]
Yeah, it would definitely be bad to store *credentials*, but that's not necessary. All it would need to store would be a list of URLs. Some people might not want that, so maybe it'd need to be an option, but I think most people would be comfortable with it, especially given the improved UX.
